include/dos_system.h

```diff
@@ -55,19 +56,37 @@ class DOS_DTA;
 
 class DOS_File {
 public:
-	DOS_File():flags(0),time(0),date(0),attr(0),refCtr(0),open(false),name(0),hdrive(0xff) { };
-	DOS_File(const DOS_File& orig);
-	DOS_File & operator= (const DOS_File & orig);
-	virtual	~DOS_File(){if(name) delete [] name;};
+	DOS_File()
+	        : flags(0),
+	          time(0),
+	          date(0),
+	          attr(0),
+	          refCtr(0),
+	          open(false),
+	          name(""),
+	          hdrive(0xff)
+	{}
+
+	DOS_File(const DOS_File &orig) = default;
+	DOS_File &operator=(const DOS_File &orig);
+
+	virtual ~DOS_File() = default;
+
+	const char *GetName() const { return name.c_str(); }
```

Comment:
Roughly a quarter of the caller of this function re-package the array back into a `std::string`; but that's a huge change :-)  One day we'll have real `string`s throughout (except for the places we feed the DOS-side with the char-array).  Maybe we'll be on Rust by then :-)